File: api/models/Museum.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const MuseumSchema = Schema({
    name: { type: String, required: true },
    city: { type: String, required: true },
    image_url : { type: String, default: "/museum/default.png"}, 
    latitude: { type: Number, required: true },
    longitude: { type: Number, required: true }
})

module.exports = mongoose.model("Museum", MuseumSchema)

File: app/lib/models/museum.dart
/*
 * Museum data model constructor
*/
class Museum {
  final String? id;
  final String name;
  final String city;
  final double latitude;
  final double longitude;

  Museum(
      {this.id,
      required this.name,
      required this.city,
      required this.latitude,
      required this.longitude});

  static Museum fromJson(Map json) => Museum(
      id: json["_id"],
      name: json["name"],
      city: json["city"],
      latitude: json["latitude"],
      longitude: json["longitude"]);
}
